src/js/components/sharedComponents/InfoContent.tsx

```diff
@@ -1,33 +1,175 @@
 import React, { FunctionComponent, useEffect, useState } from 'react';
 import { useSelector } from 'react-redux';
+import convert from 'xml-js';
 
 import { RootState } from 'js/store';
 
 import { downloadData } from '../../../../configs/modal.config';
 
+interface MetadataContent {
+  title?: string;
+  functionOrPurpose?: string;
+  geographicCoverage?: string;
+  tags?: string;
+  overview?: string;
+}
+
+interface ContentError {
+  error: string;
+}
+
+type Content = MetadataContent | ContentError;
+
+const getMetadata = async (layer: any): Promise<any> => {
+  let content: any;
+  const metadataURL = `https://www.arcgis.com/sharing/rest/content/items/${layer.portalItemID}/info/metadata/metadata.xml`;
+  const layerMetadataURL = `${layer.url}/info/metadata`;
+  const xmlResults = await fetch(metadataURL).then(res => res.text());
+  const layerXMLResults = await fetch(layerMetadataURL).then(res => res.text());
+  const metadataExists = !xmlResults.includes('Error');
+  const layerMetadataExists =
+    !layerXMLResults.includes('Error') &&
+    !layerXMLResults.includes('invalid request');
+
+  if (metadataExists) {
+    const results = convert.xml2js(xmlResults);
+    const element = results.elements[0].elements.find(
+      (element: any) => element.name === 'dataIdInfo'
+    );
+
+    const title = element.elements
+      .find((element: any) => element.name === 'idCitation')
+      .elements.find((subElement: any) => subElement.name === 'resTitle')
+      .elements[0].text;
+    const overview = element.elements.find(
+      (element: any) => element.name === 'idAbs'
+    ).elements[0].text;
+    const functionOrPurpose = element.elements.find(
+      (element: any) => element.name === 'idPurp'
+    ).elements[0].text;
```

Comment:
we can extract this into a function or something like that because we are using this exact code multiple times?